InterGalacticCurrencyConvertor/src/main/java/DataProcessResult.java

```diff
@@ -0,0 +1,44 @@
+package main.java;
+
+public class DataProcessResult{
+	private String mOutput;
+	private Boolean mSuccess;
+
+	public DataProcessResult () {
+		mOutput = null;
+		mSuccess = false;
+	}
+	public void setOutput(String inOutput) {
+		this.mOutput = inOutput;
+	}
+	public void setSuccess(Boolean inSuccess) {
+		this.mSuccess = inSuccess;
+	}
+	public String getOutput() {
+		return mOutput;
+	}
+	public Boolean hasSucceeded() {
+		return mSuccess;
+	}
+	@Override
+	public boolean equals(Object inObj) {
+		if (inObj == null) {
```

Comment:
The equals code is overly complex. Consider something like this:

// Note: this handles when other == null as well
if (!(other instanceof DataProcessResult)) {
  return false;
}

// Same instance
if (other == this) {
  return true;
}

DataProcessResult otherObj = (DataProcessResult) other;
// More details about Objects.equals here: https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html
return Objects.equals(this.mOutput, otherObj.mOutput)
    && Objects.equals(this.mSuccess, otherObj.mSuccess);

Reply:
updated